DOCS 522: improve sensor api and add model configuration pages (#981)
viamrobotics/docs: docs/components/sensor/ultrasonic.md

---
title: "Configure an ultrasonic Sensor"
linkTitle: "ultrasonic"
weight: 60
type: "docs"
description: "Configure an ultrasonic model sensor."
tags: ["sensor", "components"]
icon: "img/components/sensor.png"
# SME: #team-bucket
---

Configure an `ultrasonic` sensor to integrate an [HC-S204 ultrasonic distance sensor](https://www.sparkfun.com/products/15569) into your robot.

Configure a `ultrasonic` sensor as follows:

{{< tabs >}}
{{% tab name="JSON Template" %}}

On the **COMPONENTS** subtab of your robot's page in [the Viam app](https://app.viam.com), navigate to the **Create Component** menu.
Enter a name for your sensor, select the type `sensor`, and select the `ultrasonic` model.

```json {class="line-numbers linkable-line-numbers"}
{
  "components": [
    {
      "name": <your-ultrasonic-sensor-name>,
      "type": "sensor",
      "model": "ultrasonic",
      "attributes": {
        "trigger_pin": <number>,
        "echo_interrupt_pin": <your-digital-interrupt-name>,
        "board": <your-board-name>,
        "timeout_ms": <number>
      },
      "depends_on": []
    },

>   {

Reviewer: I'll leave the accuracy questions to @cheukt and @npmenard.

Can you move the sentence "You must [configure a digital interrupt](/components/board/#digital-interrupts) as part of your [board's configuration](/components/board/#configuration) to configure this sensor." to above the tabs?

>   {

Reviewer 2: the SME on this matter would be @npmenard and the RDK team, so unfortunately I won't be much help here - just commenting so you guys don't wait for my comments

>   {

Author: @cheukt No worries, will remove your request for review but thanks for taking a look anyways. 

and @npmenard thank you that is a great explanation! 

@npentrel Decided to delete that sentence as inaccurate

      },
      "depends_on": []
    }
  ]
}
```

{{% /tab %}}
{{% /tabs %}}

You must configure a digital interrupt as part of your [board's configuration](/components/board/#configuration) to configure this sensor.
See [here](/components/board/#digital-interrupts) for instructions on how to modify the Template or Example JSON to do so.

| Attribute | Inclusion | Description |
| ----------- | -------------- | --------------  |
| `trigger_pin` | **Required** | The pin number on the board that you have connected [the sensor's trigger pin](https://www.sparkfun.com/products/15569). Example: `"5"`. |
| `echo_interrupt_pin` | **Required** | The `name` of the [digital interrupt](/components/board/#digital-interrupts) you have configured on your board that contains the pin number of the pin [the sensor's echo pin](https://www.sparkfun.com/products/15569) is connected to. Example: `"my-interrupt-1"`. |
| `board`  | **Required** | The `name` of the [board](/components/board) the sensor is connected to. |
| `timeout_ms`  | Optional | Time to wait in milliseconds before timing out of requesting to get readings from the sensor. Default: `1000`. |

{{% alert title="Tip" color="Tip" %}}

> An `trigger_pin` value of `5` would correspond to [Pin 5 GPIO 3](https://pinout.xyz/pinout/pin5_gpio3) on a Raspberry Pi.

Reviewer: ```suggestion
A `trigger_pin` value of `5` corresponds to [Pin 5 GPIO 3](https://pinout.xyz/pinout/pin5_gpio3) on a Raspberry Pi.
```

Author: Done!

---
title: "Configure an ultrasonic Sensor"
linkTitle: "ultrasonic"
weight: 60
type: "docs"
description: "Configure an ultrasonic model sensor."
tags: ["sensor", "components"]
icon: "img/components/sensor.png"
# SME: #team-bucket
---

Configure an `ultrasonic` sensor to integrate an [HC-S204 ultrasonic distance sensor](https://www.sparkfun.com/products/15569) into your robot:

{{< tabs >}}
{{% tab name="Config Builder" %}}

> On the **COMPONENTS** subtab of your robot's page in [the Viam app](https://app.viam.com), navigate to the **Create Component** menu.Enter a name for your sensor, select the type `sensor`, and select the `ultrasonic` model.

Reviewer: [nit]
```suggestion
On the **COMPONENTS** subtab of your robot's page in [the Viam app](https://app.viam.com), navigate to the **Create Component** menu.
Enter a name for your sensor, select the type `sensor`, and select the `ultrasonic` model.
```

Paste into the **Attributes** box:

``` json
{

>   "trigger_pin": <"number">,
>   "echo_interrupt_pin": <"number">,

Reviewer: ```suggestion
  "trigger_pin": "<number>",
  "echo_interrupt_pin": "<number>"
```
